From 970ac69a935d876a0d8311620d29b8439a6320ee Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:04:58 -0400
Subject: [tools] perf/callchain: Do not report zero address in unwind

Message-id: <1410181752-25631-54-git-send-email-jolsa@redhat.com>
Patchwork-id: 91736
O-Subject: [PATCH RHEL7.1 BZ1134356 053/307] perf callchain: Do not report zero address in unwind
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 1cf0382af98f6365b01b59453fe18dffe3c73d2f
Author: Jiri Olsa <jolsa@redhat.com>
Date: Tue Jan 7 13:47:24 2014 +0100

description
===========
We are not interested in zero addresses in callchain, do not report
them.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/unwind.c b/tools/perf/util/unwind.c
index bff3209..3b70181 100644
--- a/tools/perf/util/unwind.c
+++ b/tools/perf/util/unwind.c
@@ -563,7 +563,7 @@ static int get_entries(struct unwind_info *ui, unwind_entry_cb_t cb,
   unw_word_t ip;
 
   unw_get_reg(&c, UNW_REG_IP, &ip);
-  ret = entry(ip, ui->thread, ui->machine, cb, arg);
+  ret = ip ? entry(ip, ui->thread, ui->machine, cb, arg) : 0;
  }
 
  unw_destroy_addr_space(addr_space);
-- 
1.7.1